### DISCUSSION TITLE
FulfillmentOutbound Access denied errors

### DISCUSSION THREAD

**Discussion by dave-sewell** [abuzuhri/Amazon-SP-API-CSharp#315] [Q&A] [answered]
I am getting "Access to requested resource is denied" errors attempting to use FulfillmentOutbound.GetFulfillmentOrder and FulfillmentOutbound.ListAllFulfillmentOrders. The same program, using the same credentials, makes calls to functions in FulfillmentInbound, Reports, Orders and FbaInventory that all work just fine. Any suggestions?

Dave

**Comment by abuzuhri** [comment]
you need to contact amazon for that ,but first check roles you give to your application

**Comment by dave-sewell** [comment]
We created the app with all permissions granted but did not request access to PII since we don't need it.
![SP-API-App-Roles](https://user-images.githubusercontent.com/111020676/189502675-2544784e-5a26-466c-af24-7f86fd2fd49f.jpg)

**Comment by abuzuhri** [comment]
try to create new refresh token and check if that work ?

**Reply by dave-sewell** [reply]
Creating a new refresh token did not work.
I also tried creating a second app. The second app had the same access denied error when calling FulfillmentOutbound functions.

**Reply by abuzuhri** [reply]
Then its better to open case in amazon and see what they answer

**Comment by dave-sewell** [comment, marked as answer]
I have opened a case with Amazon but in the meantime, I think I may have found a bug in the library that is causing this.
Take a look at "protected class FulfillmentOutboundApiUrls". The first private member _resourceBaseUrl is set to "/ba/outbound/2020-07-01". The letter "f" is missing at the front. It should be "/fba/outbound/2020-07-01". Could you publish a new version of the library with that correction?

**Reply by abuzuhri** [reply]
Ok I will do that today

**Reply by dave-sewell** [reply]
Thank you!

**Reply by abuzuhri** [reply]
Its fixed now

**Reply by dave-sewell** [reply]
Yes, 1.4.27 fixes that. Thanks!
